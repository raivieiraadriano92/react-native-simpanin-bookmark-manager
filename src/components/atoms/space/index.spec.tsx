import React from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import normalize from 'react-native-normalize'

import DefaultTheme from 'src/theme'

import Space from './index'

describe('atoms / Space', () => {
  it('should be able to render correctly', () => {
    const { findByText, getByTestId, toJSON } = render(
      <ThemeProvider theme={DefaultTheme}>
        <Space />
      </ThemeProvider>
    )

    const json = toJSON()

    const component = getByTestId('container')

    expect(json).toMatchSnapshot()

    expect(findByText('test')).toBeTruthy()

    expect(component.props.style).toMatchObject({
      height: normalize(DefaultTheme.spacing.medium, 'height'),
      width: '100%'
    })
  })

  it('should be able to render correctly with other props', () => {
    const { findByText, getByTestId, toJSON } = render(
      <ThemeProvider theme={DefaultTheme}>
        <Space size="small" type="vertical" />
      </ThemeProvider>
    )

    const json = toJSON()

    const component = getByTestId('container')

    expect(json).toMatchSnapshot()

    expect(findByText('test')).toBeTruthy()

    expect(component.props.style).toMatchObject({
      height: '100%',
      width: normalize(DefaultTheme.spacing.small)
    })
  })
})
