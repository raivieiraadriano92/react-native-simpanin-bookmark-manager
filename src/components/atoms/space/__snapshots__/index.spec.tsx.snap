// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms / Space should be able to render correctly 1`] = `
<View
  style={
    Object {
      "height": 48,
      "width": "100%",
    }
  }
  testID="container"
/>
`;

exports[`atoms / Space should be able to render correctly with other props 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "width": 32,
    }
  }
  testID="container"
/>
`;
